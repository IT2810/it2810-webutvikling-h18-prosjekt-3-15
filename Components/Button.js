import React from "react";
import {TouchableOpacity} from "react-native";

class Button extends React.Component{
    constructor(props){
        super(props);
    }

    startTimer(event) {
        let time = (this.props.isMinutes) ? this.props.time * 60 : this.props.time;
        return this.props.startTimer(time)
    }

    render() {
        return (
            <TouchableOpacity onPress={this.startTimer}>
            </TouchableOpacity>
        )
    }
}

export default Button;