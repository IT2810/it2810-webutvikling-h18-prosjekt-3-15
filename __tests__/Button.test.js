import React from 'react';
import Button from '../components/Button';
import ShallowRenderer from 'react-test-renderer/shallow';

it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Button/>);
    expect(tree).toMatchSnapshot();
});