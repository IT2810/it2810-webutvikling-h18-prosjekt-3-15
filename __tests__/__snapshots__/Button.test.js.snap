// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
/>
`;
